### PR TITLE
Scc 4334/replace owning institution

### PR DESCRIPTION
**Pull request by charmingduchess**
## Ticket:

- JIRA ticket [SCC-4334](https://newyorkpubliclibrary.atlassian.net/jira/software/c/projects/SCC/boards/126?assignee=61e836e95fcc37006878c15d&selectedIssue=SCC-4334)

## This PR does the following:

- Add owning institution to the bib details for partner records only
- modify `BibDetails.prototype.buildStandardDetail` to accommodate owner and extent instance variables, instead of only being able to build details straight from the bib. This pattern could be extended to the subject heading and other details, but maybe for later work.

## How has this been tested?

- Verifying a partner record shows up with owning institution. Can test with `pb2608686`
- Verify NYPL record does not show up with owning institution

## Accessibility concerns or updates

-

### Checklist:

- [x] I updated the CHANGELOG with the appropriate information and JIRA ticket number (if applicable).
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

[SCC-4334]: https://newyorkpubliclibrary.atlassian.net/browse/SCC-4334?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/models/BibDetails.ts [hunk 1]
```diff
@@ -43,6 +45,17 @@ export default class BibDetails {
     this.bottomDetails = this.buildBottomDetails()
   }
 
+  buildOwner(): string[] {
```
Comment (dgcohen) [inline src/models/BibDetails.ts:48]: Just curious, why does this have to be an array if it always returns one element?

File: src/models/BibDetails.ts [hunk 2]
```diff
@@ -124,17 +140,10 @@ export default class BibDetails {
           detail = this.buildInternalLinkedDetail(fieldMapping)
         else if (fieldMapping.field === "subjectLiteral")
           detail = this.subjectHeadings
-        else if (fieldMapping.field === "extent") detail = this.extent
-        else if (fieldMapping.field === "owner")
-          detail = this.bib.owner && {
```
Comment (dgcohen) [inline src/models/BibDetails.ts:129]: Where do the owner and extent attributes get called now that this has been removed?

File: src/models/BibDetails.ts (hunk 1 above)
Comment (charmingduchess) [reply src/models/BibDetails.ts:48]: I think I was trying to keep the details as standard as possible. The subject heading urls are nested arrays, and the intention was to have all the values for the bib details be string arrays.

**Comment by charmingduchess** [comment]
@dgcohen I added 2 more commits to try and use the same pattern more, as well as renaming a method.

File: src/models/BibDetails.ts (hunk 2 above)
Comment (charmingduchess) [reply src/models/BibDetails.ts:129]: Let me know if you want me to explain! It's a little tricky but I'm hoping this streamlines things a little.

**Review by EdwinGuzman — COMMENTED** [review]
I can't say I understand the business req/logic but the code looks right. Why did you have to change the return type from an object to an array of data? It seems like a general update.

File: __test__/pages/search/searchResults.test.tsx [hunk 3]
```diff
@@ -32,11 +32,10 @@ describe("Search Results page", () => {
       })
       await userEvent.click(field)
       await userEvent.click(screen.getByText("Apply filters"))
-      // This was the only way to get this test to pass. waitFor was not in fact waiting, even with same timeout.
-      setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (charmingduchess) [reply __test__/pages/search/searchResults.test.tsx:36]: not sure. it was not passing, so I changed it to waitFor, and the test started passing.

**Comment by charmingduchess** [comment]
@EdwinGuzman The methods that I updated had been returning a label and a value. I wanted to DRY out the logic that was adding the label, especially because the labels were hardcoded in some cases. I updated the property methods (ie extent, subjectLiteral, and other values that are transformed) to only return the value, and let `buildStandardDetail` add the label.

**Review by EdwinGuzman — APPROVED** [review]

**Review by dgcohen — APPROVED** [review]

**Pull request merged**